Relaxed UDS backend restrictions

slimhazard opened the pull request:
VCC issues a warning, but does not fail, if stat(UDS backend) fails with ENOENT or EACCES.

This makes it possible to start the UDS-listening peer, or set the socket file's permissions, after starting Varnish or loading VCL with a UDS backend.

If the file at the specified path does exist and is accessible, then it must be a socket, otherwise the VCL load fails.

Closes #2884

Dridi commented:
I think we may need one more test case:

- server s1 listens to a uds
- varnish makes a successful bereq to s1
- server s2 takes over s1's uds' path with a new socket
- varnish makes a successful bereq to s2

Otherwise LGTM except for a nitpick. This results in a message like this:

```
%s: Cannot stat: %s\n

# or

%s: Cannot stat: %s\n
(That was just a warning)\n
```

I'd rather have something like:

```
%s: Cannot stat: %s\n

# or

Warning: %s: Cannot stat: %s\n
```

Dridi commented:
About the nitpick, I see it's the same as other places so feel free to ignore it. The test case on the other hand I'd rather have in to make sure that a backend restarting while Varnish is running is not killing backend traffic until the next VCL reload or whatnot.

slimhazard commented:
The ``(That was just a warning)`` output was chosen to match the warnings generated when VCC checks references, finds unused backends or vcl subs, but ``vcc_err_unref`` is false.

@Dridi not saying you're wrong, but if we want to change the warning output, the other one should probably be changed as well. And maybe that's a subject for another issue?

slimhazard commented:
* Rebased to current master
* Added a test as @Dridi described
* Removed a minor superfluousness in the code (checking a variable for == 0 that is always 0 in that context)
* Force-pushed

IMO ready for merge now, but I'll wait until after bugwash.

Dridi commented on bin/varnishtest/tests/v00038.vtc:
```diff
@@ -123,16 +123,39 @@ varnish v1 -errvcl "Must be an absolute path:" {
+
+client c1 {
+	txreq
+	rxresp
```
Might want to check the status too

slimhazard commented:
Merged to master, travis is green.
